server/NetPacksServer.cpp

```diff
@@ -37,76 +37,80 @@ void ApplyGhNetPackVisitor::visitSaveGame(SaveGame & pack)
 
 void ApplyGhNetPackVisitor::visitEndTurn(EndTurn & pack)
 {
-	if (!gh.hasPlayerAt(pack.player, pack.c))
-		gh.throwAndComplain(&pack, "No such pack.player!");
-
+	gh.throwIfWrongPlayer(&pack);
 	result = gh.turnOrder->onPlayerEndsTurn(pack.player);
 }
 
 void ApplyGhNetPackVisitor::visitDismissHero(DismissHero & pack)
 {
-	gh.throwOnWrongOwner(&pack, pack.hid);
+	gh.throwIfWrongOwner(&pack, pack.hid);
 	result = gh.removeObject(gh.getObj(pack.hid));
 }
 
 void ApplyGhNetPackVisitor::visitMoveHero(MoveHero & pack)
 {
-	result = gh.moveHero(pack.hid, pack.dest, 0, pack.transit, gh.getPlayerAt(pack.c));
+	gh.throwIfWrongOwner(&pack, pack.hid);

```

Comment:
removed here, otherwise shows system error if movement is interrupted by timer, because client may receive YourTurn after TryMoveHero https://github.com/vcmi/vcmi/pull/2576/files#diff-a3482c9af9fda544fe0aaddadc25fd201705f93609f5d512899799c8ec879703

Reply:
So? Let's allow players to hack the game and move heroes of another players?

Reply:
any evidence that removing this check allows hacking game?

Reply:
Look into code if this is not obvious?

CGameHandler::moveHero does not makes any checks for hero owner. So, malicious client can send MoveHero netpack, with `pack.hid` set to hero of another player and server will happily apply this pack

Reply:
And if this result in error message then something wrong with caller code.
Why CPlayerInterface sends pack with pack.player set to something other than owner of our hero?

Reply:
you are probably right, the fix was in another place